react/AddressForm.tsx

```diff
@@ -89,49 +86,54 @@ const getExcludedFields = (
   return excludedFields
 }
 
-const hasWithoutNumberOption = (label: string) => {
-  return label.endsWith('Option')
-}
-
 interface AddressFormProps {
   hiddenFields?: AddressFields[]
   mandatoryEditableFields?: AddressFields[]
   onResetAddress?: () => void
-}
-
-interface FieldMeta {
-  blurred: boolean
-}
-
-type FieldsMeta = {
-  [field in AddressFields]?: FieldMeta
+  form?: ReturnType<typeof useAddressForm>
+  expanded?: boolean
+  onExpandForm?: () => void
 }
 
 const AddressForm: React.FC<AddressFormProps> = ({
   hiddenFields = [],
   mandatoryEditableFields = [],
   onResetAddress,
+  form: propsForm,
+  expanded: propsExpanded,
+  onExpandForm,
 }) => {
   const intl = useIntl()
-  const { address, setAddress, invalidFields, rules } = useAddressContext()
-  const [editing, setEditing] = useState(false)
+  const addressContext = useAddressContext()
+  const localForm = useAddressForm({
+    initialAddress: addressContext.address,
+    onAddressChange: addressContext.setAddress,
+  })
 
-  const countryRules = (address.country && rules[address.country]) || undefined
+  const [localExpanded, setLocalExpanded] = useState(false)
 
-  const { fields, display } = countryRules ?? {}
-  const summary = display?.extended
+  const expanded = propsExpanded ?? localExpanded
+
+  const form = propsForm ?? localForm
+  const address = propsForm?.address ?? addressContext.address
```

Comment:
Couldn't this line be 

```suggestion
  const address = propsForm?.address ?? localForm.address
```

as you did in the previous LOC?

Reply:
or just

```suggestion
  const address = form.address
```

Reply:
I can't if I don't want to introduce a breaking change here :-/

The reason for that is before this PR, we would always consume the address from the address context, right? So you could use the `useAddressContext().setAddress` function to change one of the field values "outside" this component, and if I change it to get the address from either the `form` or `localForm` instead of context, it wouldn't work anymore